### ISSUE TITLE
Bug in original apps API creates duplicates

### ISSUE THREAD

**Issue by Friedrieck** [olibos/HomeAssistant-RecycleApp#57]
As explained in this issue from another integration here https://github.com/pippyn/Home-Assistant-Sensor-Afvalbeheer/issues/367#issuecomment-1945999055, the original Recycle App API sometimes creates duplicates in the data returned, and they're not going to fix it. It also affects this integration, especially the calendar entity:
![image](https://github.com/olibos/HomeAssistant-RecycleApp/assets/12567014/a87785ba-a4ac-4842-b133-b9c51e4b0814)

Duplicated events are returned for PMC (here using `service: calendar.get_events`, see two first events):
```
calendar.ceroux:
  events:
    - start: "2024-02-15"
      end: "2024-02-16"
      summary: Céroux PMC
    - start: "2024-02-15"
      end: "2024-02-16"
      summary: Céroux PMC
    - start: "2024-02-15"
      end: "2024-02-16"
      summary: Déchets organiques
    - start: "2024-02-20"
      end: "2024-02-21"
      summary: Déchets organiques
    - start: "2024-02-15"
      end: "2024-02-16"
      summary: Ordures ménagères résiduelles
    - start: "2024-02-20"
      end: "2024-02-21"
      summary: Ordures ménagères résiduelles
```
The two first entries are duplicates and therefore repeat in the calendar view. So the integration should ideally correct the API bug by filtering out the duplicates when creating the events (preferably already when fetching the data).

**Comment by olibos** [comment]
Hello @Friedrieck , I'll have a look and see how I can dedupe thoses collects.
Thank you for reporting.

**Comment by olibos** [comment]
May I ask you an approximate address to check the api result?

**Comment by Friedrieck** [comment]
> May I ask you an approximate address to check the api result?

I answered you via a PM on the HA community forum (did not find a way to PM here...)

**Comment by olibos** [comment]
Thank you for the PM.

I didn't know about `service: calendar.get_events` 😉
I'll fix the duplicate collection and also sort results per date.

Here is a short preview:
Request:
```yml
service: calendar.get_events
data:
  start_date_time: "2024-02-15 00:00:00"
  end_date_time: "2024-02-20 00:00:00"
target:
  entity_id: calendar.ceroux
```

Response:
```yml
calendar.ceroux:
  events:
    - start: "2024-02-15"
      end: "2024-02-16"
      summary: PMC
    - start: "2024-02-15"
      end: "2024-02-16"
      summary: Déchets organiques
    - start: "2024-02-15"
      end: "2024-02-16"
      summary: Ordures ménagères résiduelles
    - start: "2024-02-20"
      end: "2024-02-21"
      summary: Déchets organiques
    - start: "2024-02-20"
      end: "2024-02-21"
      summary: Ordures ménagères résiduelles
```

**Comment by olibos** [comment]
It should be fixed in [v2.3.1](https://github.com/olibos/HomeAssistant-RecycleApp/releases/tag/v2.3.1)
Can you have a look?

**Comment by Friedrieck** [comment]
> It should be fixed in [v2.3.1](https://github.com/olibos/HomeAssistant-RecycleApp/releases/tag/v2.3.1)
> Can you have a look?

It seems fixed, thanks!
